onionshare_gui/threads.py

```diff
@@ -47,29 +59,28 @@ def run(self):
         self.mode.web.generate_static_url_path()
 
         # Choose port and password early, because we need them to exist in advance for scheduled shares
-        self.mode.app.stay_open = not self.mode.common.settings.get(
-            "close_after_first_download"
-        )
         if not self.mode.app.port:
             self.mode.app.choose_port()
-        if not self.mode.common.settings.get("public_mode"):
+        if not self.mode.settings.get("general", "public"):
             if not self.mode.web.password:
                 self.mode.web.generate_password(
-                    self.mode.common.settings.get("password")
+                    self.mode.settings.get("persistent", "password")
                 )
 
         try:
             if self.mode.obtain_onion_early:
                 self.mode.app.start_onion_service(
-                    await_publication=False, save_scheduled_key=True
+                    self.mode.settings, await_publication=False, save_scheduled_key=True
                 )
                 # wait for modules in thread to load, preventing a thread-related cx_Freeze crash
                 time.sleep(0.2)
                 self.success_early.emit()
                 # Unregister the onion so we can use it in the next OnionThread
-                self.mode.app.onion.cleanup(False)
+                self.mode.app.start_onion_service(self.mode.settings)
```

Comment:
Can we clarify why we're calling `start_onion_service` when we're meant to be unregistering the onion?

Reply:
I think that's why I get the error 'There was an error with Tor: ADD_ONION response didn't have an OK status: Onion address collision' when trying to schedule a share. We need to unregister the onion here so we can re-use it later for a scheduled share